spacy/language.py

```diff
@@ -922,7 +1223,70 @@ def _link_components(self):
                     if hasattr(proc2, "model"):
                         proc1.find_listeners(proc2.model)
 
-    def to_disk(self, path, exclude=tuple()):
+    @classmethod
+    def from_config(
+        cls,
+        config: Union[Dict[str, Any], Config] = {},
+        disable: Iterable[str] = tuple(),
+        overrides: Dict[str, Any] = {},
+        auto_fill: bool = True,
+        validate: bool = True,
+    ) -> "Language":
+        """Create the nlp object from a loaded config. Will set up the tokenizer
+        and language data, add pipeline components etc. If no config is provided,
+        the default config of the given language is used.
+        """
+        if auto_fill:
+            config = util.deep_merge_configs(config, cls.default_config)
+        if "nlp" not in config:
+            raise ValueError(Errors.E985.format(config=config))
+        nlp_config = config["nlp"]
+        config_lang = nlp_config["lang"]
+        if cls.lang is not None and config_lang is not None and config_lang != cls.lang:
+            raise ValueError(
+                Errors.E958.format(
+                    bad_lang_code=nlp_config["lang"],
+                    lang_code=cls.lang,
+                    lang=util.get_object_name(cls),
+                )
+            )
+        nlp_config["lang"] = cls.lang
+        # This isn't very elegant, but we remove the [pipeline] block here to prevent
+        # it from getting resolved (causes problems because we expect to pass in
+        # the nlp and name args for each component). If we're auto-filling, we're
+        # using the nlp.config with all defaults.
+        config = Config().from_str(config.to_str())  # deepcopy config!
+        orig_pipeline = config["nlp"].pop("pipeline", {})
+        config["nlp"]["pipeline"] = {}
+        non_pipe_overrides, pipe_overrides = _get_config_overrides(overrides)
+        resolved, filled = registry.resolve(
+            config, validate=validate, schema=ConfigSchema, overrides=non_pipe_overrides
+        )
+        filled["nlp"]["pipeline"] = orig_pipeline
+        config["nlp"]["pipeline"] = orig_pipeline
+        create_tokenizer = resolved["nlp"]["tokenizer"]
+        lemmatizer = resolved["nlp"]["lemmatizer"]
+        nlp = cls(create_tokenizer=create_tokenizer, lemmatizer=lemmatizer)
```

Comment:
I do see why you were eager to avoid this approach and instead let the make from config handle it, but I think in the end this is quite okay. Yeah it's a bit fiddly, but it's one bit of code and you've made it quite easy to understand imo. I like that you can see the `add_pipe` loop in there, just like you'd have in a script.

Reply:
Yeah, I think this is okay 🙂 Also check out the `Vocab.from_config` – if we adjust that and improve the handling of defaults, move the lemmatizer out etc., this can actually be pretty elegant.